src/Microsoft.Performance.SDK.Runtime/Extensibility/BaseSourceProcessingSession.cs

```diff
@@ -190,15 +192,16 @@ protected void InitializeForSourceParsing()
         protected void ClearActiveCookers(int newCapacityCount)
         {
             this.activeCookers = new List<ISourceDataCooker<T, TContext, TKey>>(newCapacityCount);
-            this.activeReceiveAllDataCookers = new HashSet<ISourceDataCooker<T, TContext, TKey>>();
+            this.activeReceiveAllDataCookers = new List<ISourceDataCooker<T, TContext, TKey>>();
         }
 
         protected void ActivateCooker(ISourceDataCooker<T, TContext, TKey> sourceDataCooker)
         {
             Guard.NotNull(sourceDataCooker, nameof(sourceDataCooker));
 
             this.activeCookers.Add(sourceDataCooker);
-            if (sourceDataCooker.Options.HasFlag(SourceDataCookerOptions.ReceiveAllDataElements))
+            if (sourceDataCooker.Options.HasFlag(SourceDataCookerOptions.ReceiveAllDataElements) &&
+                !this.activeReceiveAllDataCookers.Contains(sourceDataCooker))
```

Comment:
this might be expensive. is this a hot path?

Reply:
It could be expensive, but should not be a problem as it isn't a hot path.